### ISSUE TITLE
Shift e2e processing into a webworker

### ISSUE THREAD

**Issue by ara4n** [vector-im/vector-web#2503]
We shouldn't be locking up the UI when encrypting/decrypting E2E; shouldn't this be in a bg worker?

**Comment by richvdh** [comment]
Yes it should, but that will mean asynchronous decryption of events, which will probably mean hooks up to the react layer.

**Comment by ara4n** [comment]
now the whole API is async thanks to indexeddb, does this become easier?  Or have we even got it for free?

**Comment by marcusmueller** [comment]
Since this has become the blocker issue for #26231 (which is a paying-customer facing usability regression), I'd like to understand what the status of this issue is, and whether we are "close" in temporal terms.

**Comment by richvdh** [comment]
@marcusmueller I'm afraid it's unlikely to be a priority within the next few weeks.

**Comment by marcusmueller** [comment]
Now, I understand that weeks are short. I feel like I'm touting my own horn here a bit hard, but if this change does not end up being a priority at some point in the medium future, that'd be a bit sad:<br>
We have a significant UX regression, which was brought by the necessity of being safe around rust-sdk-crypto not being concurrency-safe, and the plan would be neither to fundamentally change rust-sdk nor element-web to actually work around this fact, but live with the regression that users can no longer use the same session in multiple tabs.

**Comment by lunera-dev** [comment]
It's 7+ months later and this is still an issue.  In my use case, I am NOT using E2E because encryption in my org is against policy.  Is there a way I can have multiple tabs (rooms) open at the same time since E2E is disabled? #26231 links here.
